src/sentry/replays/query.py

```diff
@@ -184,7 +187,60 @@ def query_replays_count(
     )
 
 
-# Select.
+def query_replays_dataset_tagkey_values(
+    project_ids: List[str],
+    start: datetime,
+    end: datetime,
+    environment: str | None,
+    tag_key: str,
+):
+    """Query replay tagkey values. Like our other tag functionality, aggregates do not work here."""
+
+    query_options = {"limit": Limit(1000)}
+    where = []
+
+    if environment:
+        where.append(Condition(Column("environment"), Op.IN, environment))
+
+    value = TAG_QUERY_ALIAS_COLUMN_MAP.get(tag_key)
+    if value is None:
+        group_by_param = Function(
+            "arrayJoin",
```

Comment:
Do we know how this affects the limit semantics?  If there are 1000 tag values per row does that yield 1000 * 1000 rows in the result?  Not sure if that's the case or what can be done.  Just an observation.  When we build the response dictionary you might want to cap the data key (e.g. `for d in results["data"][:1000]`) so we don't flood the client with values.

Reply:
confirmed via local clickhouse query that this does not effect limit semantics